From c79925642585e498c8d3e23a6a179599be6e822b Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 6 Jun 2019 01:51:50 +0200
Subject: [PATCH 1/1] efi_loader: SignalEvent for event in signaled state

If an event is already in the signaled state, SignalEvent should not queue
the notification function but simply return EFI_SUCCESS.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 54fff85e64..90842e7664 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -221,6 +221,8 @@ efi_status_t is_valid_tpl(efi_uintn_t tpl)
  */
 void efi_signal_event(struct efi_event *event, bool check_tpl)
 {
+	if (event->is_signaled)
+		return;
 	if (event->group) {
 		struct efi_event *evt;
 
-- 
2.20.1

